### PR TITLE
[feat] Improve flexibility for BOATs integration

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

- Removed `DataBaseHandler` class and instead use `sqlalchemy.Connection` objects for more flexibility
- Added CLI help menu and config example to docs, see [updated docs](https://cafntem--29.org.readthedocs.build/en/29/)

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases) - No unit tests but integrations tests have been updated
- [x] Has documentation (including user guide) been updated - yes
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt` - added pydantic
- [x] Code linting, tests and other workflows pass

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -7,9 +7,10 @@
 Welcome to CAF.ntem's documentation!
 ===========================================================
 
-CAF package for extracting and analysing NTEM data.
-
-CAF.ntem source code is available on `GitHub <https://github.com/transport-for-the-north/caf.ntem>`_.
+Common analytical framework (CAF) Python package for extracting and analysing NTEM data.
+CAF.ntem provides functionality to load NTEM datasets into a database and
+query the database to produce planning, trip end and car ownership datasets. The
+tools functionality is provided via a command-line interface (CLI) and Python API.
```
Comment (BenTaylor-TfN) [inline docs/source/index.rst]:
```suggestion
Common analytical framework (CAF) Python package for extracting and analysing NTEM data.

NTEM (National Trip End Model) is a UK Department for Transport (DfT) model that forecasts
trip end data up to 2051. More information and the data from this model can be found on 
[data.gov.uk](https://www.data.gov.uk/dataset/11bc7aaf-ddf6-4133-a91d-84e6f20a663e/national-trip-end-model-ntem).

CAF.ntem allows easy querying of this dataset through Python, allowing users to extract
planning, trip end and car ownership datasets. CAF.NTEM provides both a 
command-line interface (CLI) and Python API.

```

**Review by BenTaylor-TfN — COMMENTED** [review]
Code changes look good to me, all sensible changes for sqlalchemy. I've added one suggested change for the docs, mostly to describe NTEM and to link to the source so people know where / how to get the data. Not sure if this is mentioned anywhere else?

**Comment by MattBuckley-TfN** [comment]
Added optional parameter to query method which outputs zone name column in addition to zone code

**Review by BenTaylor-TfN — APPROVED** [review]
